### ISSUE TITLE
[Machine Learning] Building a YOLACT model using MM Detection

### ISSUE THREAD

**Issue by srishilesh** [section-engineering-education/engineering-education#6994]
### Topic Suggestion
Machine Learning

### Writing sample(s): 
https://www.section.io/engineering-education/authors/srishilesh-p-s/

#### Proposed title of article
Building a YOLACT model using MM Detection

#### Proposed article introduction
In this tutorial, you will learn about what MM detection is, how it can leveraged to build object detection models easily with more accuracy. Building an object detection model ground-up using frameworks like Tensorflow or PyTorch is difficult. With MM detection, the developer will use certain APIs to build a deep learning model in less 5 lines.

We will learn the architectural basics of YOLACT model and understand how MM detection works. Later, we will build deep learning model to detect table columns from a table dataset from scratch.

#### Key takeaways 
On reading this tutorial, the reader will understand:
- The basics of MM detection and how to work with it
- The basics of YOLACT architecture
- How to call MM detection APIs to train-test models
- How to visualize and analyze using the metric APIs

#### Article quality
MM detection is an open source library. We don't find many articles that tells about the working of MM detection. MaskRCNN using MM detection has been the most covered article on the internet. While building MaskRCNN model is implicitly easier, many factors while building other models are not considered. The developers find it hard to look through the poor documentation. Hence, this article will act as a guide to build their deep learning models easily.

#### References
- https://github.com/open-mmlab/mmdetection
- https://github.com/dbolya/yolact

**Comment by srishilesh** [comment]
Hey @WanjaMIKE ,

Why was this issue closed?

**Comment by WanjaMIKE** [comment]
Unexisting PR @srishilesh

**Comment by srishilesh** [comment]
> Unexisting PR @srishilesh

Hey @WanjaMIKE ,

I know that we were having a rule of submitting 1 topic suggestion per author and wait until it's linked PR gets merged. Are we still following the same rule?

Didn't we bring in this rule to avoid reviews on multiple PRs (containing major revisions) by the same author? I thought this rule was applicable only to the new contributors or for contributors who are just getting started with technical writing.

Shouldn't we increase the topic suggestions count for other type of contributors (who have been contributing at Section for a while)? I remember reading one such comment in our peer-reviewers channel a while ago. Not sure if this rule is in place now.

Any thoughts? @hectorkambow

**Comment by WanjaMIKE** [comment]
I think its still applicable. Could you text Hector about it. Thanks

**Comment by srishilesh** [comment]
> I think its still applicable. Could you text Hector about it. Thanks

That's great then. I'll ping Hector about this.

Thanks for the reply.

**Comment by hectorkambow** [comment]
> > Unexisting PR @srishilesh
> 
> Hey @WanjaMIKE ,
> 
> I know that we were having a rule of submitting 1 topic suggestion per author and wait until it's linked PR gets merged. Are we still following the same rule?
> 
> Didn't we bring in this rule to avoid reviews on multiple PRs (containing major revisions) by the same author? I thought this rule was applicable only to the new contributors or for contributors who are just getting started with technical writing.
> 
> Shouldn't we increase the topic suggestions count for other type of contributors (who have been contributing at Section for a while)? I remember reading one such comment in our peer-reviewers channel a while ago. Not sure if this rule is in place now.
> 
> Increasing topic suggestions limit for active contributors will definitely boost the PR count.
> 
> Any thoughts? @hectorkambow

Thank you for the great suggestion @srishilesh  

This policy was put in place more to help us with the influx of articles coming in (as well). 

Since we have a small team of PR & CMs. 

Allowing multiple PRs by authors would increase our queue and our publishing rate would slow down.

**Comment by srishilesh** [comment]
@hectorkambow 

Thanks for the clarification. You're right. If we have multiple PRs by the same author, it might slow down the publishing rate.

So, can't we get the next topic suggestion reviewed until the current PR is in review?

**My suggestion:**
At any point of time, we may have at max. 2 topic suggestions open, while the author is allowed to raise only 1 PR at a time, such that the PR is ready for final review.

**Pros:**
- Reduces the time gap in waiting to get the PR merged and waiting to get approval for the next topic suggestion.
- No increase in the PR queue - since no multiple PRs.
- Faster PR submission process for the author. Can avoid excessive time wastage on approvals/merging and can focus on writing articles.
- The publishing rate wouldn't be affected.
- We can get more active contributors (authors). If you notice the article publishing (PR merging) cycle of any active contributor, it would be an approximate of 1 - 2 months i.e. the throughput time to get an article published.

**Cons:**
- Increase in topic suggestions.
- Need to keep a check on authors if they have multiple PRs in the queue.

(These are just my opinions. We may ask the bigger team to refine these ideas.)

Your thoughts?

**Comment by srishilesh** [comment]
> > So, can't we get the next topic suggestion reviewed until the current PR is in review?
> 
> Just to follow up on this line.
> 
> Authors should only have the Topic suggestion that links to the open PR. 1 for 1 - one topic suggestion & on PR open (that are the same topic and article).
> 
> & until that PR is merged (published) then a student can submit another one to be approved. @srishilesh
> 
> Not sure if that is what you meant.

Got it

**Comment by srishilesh** [comment]
Hi @WanjaMIKE ,

Can this topic be reviewed now?

My previous topic's PR is merged.

**Comment by WanjaMIKE** [comment]
Seems like a helpful topic. Lets ensure that it adds value to the community. Thanks

**Comment by hectorkambow** [comment]
Closing this one due to inactivity - will need to be re-submitted in order to proceed.
